web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/CitizenCertificateTextServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package se.inera.intyg.intygstjanst.web.service.impl;
+
+import org.springframework.stereotype.Service;
+import se.inera.intyg.common.support.modules.registry.IntygModuleRegistry;
+import se.inera.intyg.common.support.modules.registry.ModuleNotFoundException;
+import se.inera.intyg.intygstjanst.web.service.CitizenCertificateTextService;
+
+@Service
+public class CitizenCertificateTextServiceImpl implements CitizenCertificateTextService {
+
+    private final IntygModuleRegistry intygModuleRegistry;
+
+    public CitizenCertificateTextServiceImpl(IntygModuleRegistry intygModuleRegistry) {
+        this.intygModuleRegistry = intygModuleRegistry;
+    }
+
+    @Override
+    public String getTypeName(String typeId) throws ModuleNotFoundException {
+        return intygModuleRegistry.getModuleEntryPoint(typeId).getModuleName();
+    }
+
+    @Override
+    public String getAdditionalInfoLabel(String typeId, String typeVersion) throws ModuleNotFoundException {
```

Comment:
I know this is not ready yet, but this is a comment as a reminder for myself :-)

Reply:
Can we extract this fix from this PR so I can merge the changes anyways?

Reply:
Good suggestion! We can take the additionalInfoLabel as a separate task.